vmdb/app/models/mixins/retirement_mixin.rb

```diff
@@ -74,6 +74,7 @@ def retire(options = {})
       end
     end
 
+    self.retirement_requester = nil
```

Comment:
@mkanoor Resetting `retirement_requester` should be done in the methods `retires_on=` and `retirement_warn=` that are called from here.  Direct callers to those methods will want to reset that value as well.